packages/repo/src/car.ts

```diff
@@ -101,35 +108,40 @@ export const readCarStream = async (
     }
     return {
       roots: header.roots,
-      blocks: readCarBlocksIter(reader),
+      blocks: readCarBlocksIter(reader, opts),
     }
   } catch (err) {
     await reader.close()
     throw err
   }
 }
 
-const readCarBlocksIter = (reader: BufferedReader) => {
-  const iter = readCarBlocksIterGenerator(reader) as CarBlockIterable
-
-  iter.dump = async () => {
-    // try/finally to ensure that reader.close is called even if blocks.return throws.
-    try {
-      // Prevent the iterator from being started after this method is called.
-      await iter.return()
-    } finally {
-      // @NOTE the "finally" block of the async generator won't be called
-      // if the iteration was never started so we need to manually close here.
-      await reader.close()
-    }
+const readCarBlocksIter = (
+  reader: BufferedReader,
+  opts?: ReadCarOptions,
+): CarBlockIterable => {
+  let generator = readCarBlocksIterGenerator(reader)
+  if (opts?.verifyCids) {
```

Comment:
I would default this to `true`:

```suggestion
  if (opts?.verifyCids !== false) {
```

We could also name this `skipCidVerification` to make it more obvious that it needs to be set for cid verification to be disabled.

Reply:
This would also help with backwards compatibility: current dependents might implicitly assume that cids are being validated.